File: src/test/com/natpryce/snodge/IdentityMutatorTest.java
package com.natpryce.snodge;

import org.junit.Test;

import static com.google.common.collect.Lists.newArrayList;
import static java.util.Collections.singletonList;
import static org.hamcrest.Matchers.equalTo;
import static org.hamcrest.junit.MatcherAssert.assertThat;

public class IdentityMutatorTest {
    @Test
    public void createsASingleNonMutationOfOriginal() {
        Mutator<Integer> m = IdentityMutator.id();

        int original = 8;
        int mutantCount = 100;

        assertThat("mutations",
                newArrayList(m.mutate(original, mutantCount)), equalTo(singletonList(original)));
    }
}

File: src/test/com/natpryce/snodge/IdentityMutatorTest.kt
package com.natpryce.snodge

import com.natpryce.hamkrest.assertion.assertThat
import com.natpryce.hamkrest.equalTo
import org.junit.Test

class IdentityMutatorTest {
    @Test
    fun createsASingleNonmutationOfOriginal() {
        val m = id<Int>()
        
        val original = 8
        
        val actualMutations: List<Int> = m.invoke(original, 100)
        val expectedMutations : List<Int> = listOf(original)
        
        assertThat("mutations", actualMutations, equalTo(expectedMutations))
    }
}
